### PR TITLE
Upgrade EF to 2.0.0-rtm

### LINKED ISSUE #317: Upgrade to MySqlConnector that Supports netstandard2.0 Before 2.0.0-RTM
MySqlConnector will publish a `netstandard2.0` build once 2.0.0 RTM is published

We need to wait for this to get published on NuGet, then switch our dependencies to use it before we publish our 2.0.0 package.

Here is the tracking issue: https://github.com/mysql-net/MySqlConnector/issues/270

### LINKED ISSUE #350: Upgrade to 2.0.0-rtm

### PR DESCRIPTION
**Pull request by caleblloyd**
- Fixes #350 
- Fixes #317 
- ~~Cannot fix tests until next version of .NET Core 2.0 is published~~ it is published

### CODE REVIEW

**Comment by Kagamine** [comment]
https://www.microsoft.com/net/download/core

File: src/EFCore.MySql/Storage/Internal/MySqlDatabaseCreator.cs [hunk 1]
```diff
@@ -95,8 +96,16 @@ private bool Exists(bool retryOnNotExists)
...
-                            _connection.DbConnection.Close();
+                            using (var masterConnection = _connection.CreateMasterConnection())
+                            {
+                                masterConnection.Open();
+                                using (var cmd = masterConnection.DbConnection.CreateCommand())
+                                {
+                                    cmd.CommandText = $"USE `{_connection.DbConnection.Database}`";
+                                    cmd.ExecuteNonQuery();
+                                }
+                                masterConnection.Close();
```
Comment (Kagamine) [inline src/EFCore.MySql/Storage/Internal/MySqlDatabaseCreator.cs]: I dont think we need close this manually

File: test/EFCore.MySql.FunctionalTests/AppDb.cs [hunk 2]
```diff
@@ -7,11 +7,6 @@
 using Microsoft.EntityFrameworkCore.Infrastructure;
 using Microsoft.Extensions.Logging;
 using Pomelo.EntityFrameworkCore.MySql.FunctionalTests.Models;
-#if ORACLE
-using MySQL.Data.EntityFrameworkCore.Extensions;
-#elif SAPIENT
```
Comment (Kagamine) [inline test/EFCore.MySql.FunctionalTests/AppDb.cs]: Remove them since they dont support 2.0

**Comment by caleblloyd** [comment]
@Kagamine scaffolding test is failing with:

```
Unable to identify the primary key for table 'DataTypesSimple'.
Unable to generate entity type for table 'DataTypesSimple'.
Unable to identify the primary key for table 'DataTypesVariable'.
Unable to generate entity type for table 'DataTypesVariable'.
```

Any idea why?

**Comment by mguinness** [comment]
@caleblloyd Just a thought, what if you add the [Key] attribute to the Id member in [DataTypes.cs](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/master/test/EFCore.MySql.FunctionalTests/Models/DataTypes.cs#L9)?  Haven't been able to test, but it might be worth trying.

**Comment by caleblloyd** [comment]
@mguinness that wasn't it, needed a new method to find PKs ac8709660b74ea2259eaeb621ad3d644b7f72b97

**Pull request merged**